symbolic-ppdb/src/sequence_points.rs

```diff
@@ -0,0 +1,313 @@
+            start_column: 0,
+            end_line: 0xfeefee,
+            end_column: 0,
+            document_id,
+        }
+    }
+
+    fn parse(
+        data: &[u8],
+        prev: Option<SequencePoint>,
+        prev_non_hidden: Option<SequencePoint>,
+        document_id: u32,
+    ) -> Result<(Self, &[u8]), Error> {
+        let (il_offset, data) = match prev {
+            Some(prev) => {
+                let (delta_il_offset, data) = decode_unsigned(data)?;
+                (prev.il_offset + delta_il_offset, data)
+            }
+            None => decode_unsigned(data)?,
+        };
+
+        let (delta_lines, data) = decode_unsigned(data)?;
+        let (delta_cols, data): (i32, &[u8]) = if delta_lines == 0 {
+            let (n, data) = decode_unsigned(data)?;
+            (n.try_into().unwrap(), data)
+        } else {
+            decode_signed(data)?
+        };
+
+        if delta_lines == 0 && delta_cols == 0 {
+            return Ok((Self::new_hidden(il_offset, document_id), data));
+        }
+
+        let (start_line, data) = match prev_non_hidden {
+            Some(prev) => {
+                let (delta_start_line, data) = decode_unsigned(data)?;
+                (prev.start_line + delta_start_line, data)
+            }
+            None => decode_unsigned(data)?,
+        };
+
+        let (start_column, data) = match prev_non_hidden {
+            Some(prev) => {
+                let (delta_start_col, data) = decode_unsigned(data)?;
+                (prev.start_column + delta_start_col, data)
+            }
+            None => decode_unsigned(data)?,
+        };
+
+        let end_line = start_line + delta_lines;
+        let end_column = (start_column as i32 + delta_cols) as u32;
+
+        Ok((
+            Self::new(
+                il_offset,
+                start_line,
+                start_column,
+                end_line,
+                end_column,
+                document_id,
+            )?,
+            data,
+        ))
+    }
+}
+
+impl fmt::Debug for SequencePoint {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.is_hidden() {
+            f.debug_struct("HiddenSequencePoint")
+                .field("il_offset", &self.il_offset)
+                .field("document_id", &self.document_id)
+                .finish()
+        } else {
+            f.debug_struct("SequencePoint")
+                .field("il_offset", &self.il_offset)
+                .field("start_line", &self.start_line)
+                .field("start_column", &self.start_column)
+                .field("end_line", &self.end_line)
+                .field("end_column", &self.end_column)
+                .field("document_id", &self.document_id)
+                .finish()
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub(crate) struct Document {
+    pub(crate) name: String,
+    pub(crate) lang: Language,
+}
+
+pub(crate) struct SequencePoints<'data> {
+    ppdb: PortablePdb<'data>,
```

Comment:
```suggestion
    ppdb: &'data PortablePdb<'data>,
```

I think this can avoid a copy of the `ppdb`.